hapi-fees/src/main/java/com/hedera/services/usage/crypto/CryptoUpdateMeta.java

```diff
@@ -0,0 +1,192 @@
+package com.hedera.services.usage.crypto;
+
+/*-
+ * ‌
+ * Hedera Services API Fees
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.base.MoreObjects;
+import com.hederahashgraph.api.proto.java.CryptoUpdateTransactionBody;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import static com.hederahashgraph.fee.FeeBuilder.BASIC_ENTITY_ID_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.INT_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.LONG_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.getAccountKeyStorageSize;
+
+public class CryptoUpdateMeta {
+
+	private final long keyBytesUsed;
+	private final long msgBytesUsed;
+	private final long memoSize;
+	private final long effectiveNow;
+	private final long expiry;
+	private final boolean hasProxy;
+	private final int maxAutomaticAssociations;
+	private final boolean hasMaxAutomaticAssociations;
+
+	public CryptoUpdateMeta(Builder builder) {
+		keyBytesUsed = builder.keyBytesUsed;
+		msgBytesUsed = builder.msgBytesUsed;
+		memoSize = builder.memoSize;
+		effectiveNow = builder.effectiveNow;
+		expiry = builder.expiry;
+		hasProxy = builder.hasProxy;
+		maxAutomaticAssociations = builder.maxAutomaticAssociations;
+		hasMaxAutomaticAssociations = builder.hasMaxAutomaticAssociations;
+	}
+
+	public CryptoUpdateMeta(TransactionBody cryptoUpdateTxn) {
+		var op = cryptoUpdateTxn.getCryptoUpdateAccount();
```

Comment:
This is a bug that allows misuse of `CryptoUpdateMeta`. Always provide the correct type of `TransactionBody` (`CryptoUpdateTransactionBody` in this case) and let the upstream caller take care of giving the correct type.